---
title: "Project"
output:
  github_document:
    toc: true
    toc_depth: 4
---
From the paper:

Align ChIP-seq reads to the Arabidopsis reference genome (TAIR10) by Bowtie with at most 2 mismatches Visualize results  with IGV software
Reads that appeared more than twice at the same position on the same strand were discarded to remove PCR duplication
MACS2 (Model-based Analysis of ChIP-seq) [32] was used to identify peaks using a q-value cutoff of 0.05.

Downloaded reference TAIR-10 genome from: https://support.illumina.com/sequencing/sequencing_software/igenome.html

Downloaded data from study using accession number CRA003775 from:
https://bigd.big.ac.cn/gsa

```{bash, eval=FALSE}
# upload files to remote server
scp /Users/alicezhang/Documents/400E_Assignments/Project/* alzhang_bmeg22@orca1.bcgsc.ca:/home/alzhang_bmeg22

scp -r /Users/alicezhang/Downloads/Arabidopsis_thaliana_Ensembl_TAIR10/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index alzhang_bmeg22@orca1.bcgsc.ca:/home/alzhang_bmeg22/genome


#align the input file
bowtie2 -x genome/genome -U CRR242641.fq -S input_align.sam
#align the Chip seq file
bowtie2 -x genome/genome -U CRR242640.fq -S align.sam

#convert sam files to bam and also sort them 
samtools view -S -b -h align.sam > alignPE.bam
samtools view -S -b -h alignSE.sam > alignSE.bam
# i think we should do this instead
samtools sort align.sam -o align.bam
samtools sort input_align.sam -o input_align.bam

#filter only uniquely mapped reads with at most 2 mismatches
sambamba view -F "[XS] == null and not unmapped and [NM] <=2)" -h --format=bam align.bam > filtered_align.bam
sambamba view -F "[XS] == null and not unmapped and [NM] <=2)" -h --format=bam input_align.bam > filtered_input_align.bam


#Create index to create bigwig files
samtools index -b filtered_align.bam
samtools index -b filtered_input_align.bam

#creating the bw files
bamCoverage -b filtered_align.bam -o filtered_align.bw
bamCoverage -b filtered_input_align.bam -o filtered_input_align.bw

#move files to local to view in IGV

# So i just open the terminal where i want to download files and use "." instead of directory address, makes things easier
#not a robust way, but easyyy

 scp adhir_bmeg22@orca1.bcgsc.ca:/home/adhir_bmeg22/project/filtered_align.bw .
 scp adhir_bmeg22@orca1.bcgsc.ca:/home/adhir_bmeg22/project/filtered_input_align.bw .
```

```{}
![](peaks_summary.png)
```

```{bash, eval=FALSE}
#install macs2 and call peaks
conda install -c bioconda macs2
#TODO macs2 not working for me rn for some reason
macs2 callpeak -q 0.05 -t filtered_align.bam -c filtered_input_align.bam -n peaks_align


#to convert fq file to fa for motif analysis
#install seqtk, library which will help do the conversion
conda install -c bioconda seqtk

#convert the file
seqtk seq -a CRR242640.fq > CRR242640.fa

#move bed file to local
scp adhir_bmeg22@orca1.bcgsc.ca:/home/adhir_bmeg22/project/peaks_align_summits.bed .
```

```{r}
# install.packages("igraph", type = "binary") (nonbinary didn't work for me for igraph)
library(dplyr)
library(igraph)

#install these with Bioconductor, don't update igraph
library(ChIPseeker)
library(clusterProfiler)
library(rtracklayer)
library(GenomicRanges)
library(regioneR)

bed <- as.data.frame(read.table("peaks_align_summits.bed",header = TRUE, sep="\t",stringsAsFactors=FALSE, quote=""))

gr_obj =  import("peaks_align_summits.bed")
gr_list = split(gr_obj, gr_obj$name)

covplot(gr_list, weightCol="X77.1138")

```

```{bash, eval=FALSE}





#installation 
#apart from meme

conda install -c conda-forge -c bioconda -c defaults prokka

#for this one nomodel works coz of the reason provided in the github link 
# http://seqanswers.com/forums/showthread.php?t=69769     ;) nice numbers in the link



#motif analysis required fasta files, could not really find packages to convert bed files to fasta so used this
#script found on stackexchange
#https://bioinformatics.stackexchange.com/questions/5368/how-to-convert-bed-file-to-fasta-file

perl bed2faidxsta.pl align_peaks_PE_summits.bed peaks_PE.fa
#perform motif analysis on peaks 

```
To identify possible binding motif of the WRKY33 transcription factor, the ChIP peak sequences were subjected to MEME (Multiple EM for Motif Elicitation)-ChIP 
The MEME-ChIP program uses two ab initio motif discovery algorithms: MEME [19], and DREME (Discriminative Regular Expression Motif Elicitation) [33], which uses regular expressions to search for short eukaryotic TF motifs that are missed by MEME.

In order to determine the putative functions of the target gene WRKY33, all identifed genes with ChIP-seq peaks in the upstream promoter region or the potential regulatory region downstream were subjected to annotation of the categories of ontological genes (GO) [20]. The default Fisher’s Exact Test and Benjamini-Yekutieli multiple test correction methods [34] were used to generate p-values for statistical significance and corresponding False Discovery Rate (FDR) values.

Total RNA was isolated using the Biospin Plant Total RNA Extraction kit according to the user manual (Bioer Technology; Hangzhou, China), from the pooled three-week old rosette leaves of Col and 35S:FLAG-WRKY33 plants, and 1–2 μg total RNA was used for reverse transcription, using the PrimeScript RT reagent kit (Takara Cat# RR047A). A QuantiNova SYBR Green PCR Kit was used for qPCR reactions with qPCR-specific primers. The expression levels of putative target genes were compared with Arabidopsis ACTIN genes.

```{r}
library(tidyverse)
library(microseq)
# CRR0 <- readFastq('CRR242640.fq.gz')
```


