---
title: "BMEG 400F Final Project"
output:
  github_document:
    toc: true
    toc_depth: 4
---
Downloaded reference TAIR-10 genome from: https://support.illumina.com/sequencing/sequencing_software/igenome.html

Downloaded data from study using accession number CRA003775 from:
https://bigd.big.ac.cn/gsa

```{bash, eval=FALSE}
# upload files to remote server
scp /Users/alicezhang/Documents/400E_Assignments/Project/* alzhang_bmeg22@orca1.bcgsc.ca:/home/alzhang_bmeg22

scp -r /Users/alicezhang/Downloads/Arabidopsis_thaliana_Ensembl_TAIR10/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index alzhang_bmeg22@orca1.bcgsc.ca:/home/alzhang_bmeg22/genome


#align the input file
bowtie2 -x genome/genome -U CRR242641.fq -S input_align.sam
#align the Chip seq file
bowtie2 -x genome/genome -U CRR242640.fq -S align.sam

#convert sam files to bam and also sort them 
samtools view -S -b -h -@ 32align.sam > align.bam
samtools view -S -b -h -@ 32 input_align.sam > input_align.bam

#filter only uniquely mapped reads with at most 2 mismatches
sambamba view -F "[XS] == null and not unmapped and [NM] <=2)" -h -t 32 --format=bam align.bam > filtered_align.bam
sambamba view -F "[XS] == null and not unmapped and [NM] <=2)" -h -t 32 --format=bam input_align.bam > filtered_input_align.bam

#gotta sort for the index thing
sort filtered_align.bam -o sorted_filtered_align.bam 
sort filtered_align.bam -o sorted_filtered_input_align.bam 

#Create index to create bigwig files
samtools index -b -@ 32 sorted_filtered_align.bam
samtools index -b -@ 32 sorted_filtered_input_align.bam


#creating the bw files

bamCoverage -b sorted_filtered_align.bam -o filtered_align.bw --numberOfProcessors max
bamCoverage -b sorted_filtered_input_align.bam -o filtered_input_align.bw --numberOfProcessors max

#move files to local to view in IGV

# So i just open the terminal where i want to download files and use "." instead of directory address, makes things easier
#not a robust way, but easyyy

 scp adhir_bmeg22@orca1.bcgsc.ca:/home/adhir_bmeg22/project/filtered_align.bw .
 scp adhir_bmeg22@orca1.bcgsc.ca:/home/adhir_bmeg22/project/filtered_input_align.bw .
```

```{}
![](peaks_summary.png)
```

```{bash, eval=FALSE}
#install macs2 and call peaks
conda install -c bioconda macs2
#TODO macs2 not working for me rn for some reason
macs2 callpeak -q 0.05 -t sorted_filtered_align.bam -c sorted_filtered_input_align.bam -n peaks_align 

#fasta for motif analysis
bedtools getfasta -fi ../project/whole_genome.fa -bed peaks_align_peaks.narrowPeak -fo fastaSeqs.fa



#to convert fq file to fa for motif analysis
#install seqtk, library which will help do the conversion
conda install -c bioconda seqtk

#convert the file
seqtk seq -a CRR242640.fq > CRR242640.fa

#move bed file to local
scp adhir_bmeg22@orca1.bcgsc.ca:/home/adhir_bmeg22/project/peaks_align_summits.bed .

#changing bed file and bam to fasta for meme
samtools ampliconclip -b peaks_align_summits.bed filtered_align.bam -o filtered.fa
#ran meme locally and on meme server
```

```{r echo=T, results='hide', message=FALSE}
# install.packages("igraph", type = "binary") (nonbinary didn't work for me for igraph)
library(dplyr)
library(igraph)

#install these with Bioconductor, don't update igraph
library(ChIPseeker)
library(clusterProfiler)
library(rtracklayer)
library(GenomicRanges)
library(regioneR)
library(VennDetail)

```

```{r}
bed2 <- as.data.frame(read.table("peaks_align_summits.bed",header = FALSE, sep="\t"))
res2 <- 
  lapply(split(bed2, bed2$V1), function(i){
    GRanges(seqnames = i$V1,
            ranges = IRanges(start = i$V2,
                             end = i$V3,
                             names = i$V4))
  })
covplot(res2)
```

```{r}
library(TxDb.Athaliana.BioMart.plantsmart28)
all_gene <- genes(TxDb.Athaliana.BioMart.plantsmart28)
resUnlisted <- unlist(GRangesList(res2))
hits <- findOverlaps(resUnlisted, all_gene)
```
Obtained EnsDb database from: https://github.com/wyguo/EnsDb.Athaliana.vAtRTD2/blob/master/inst/extdata/EnsDb.Athaliana.vAtRTD2.sqlite
```{r, message=FALSE}
# peak annotation
annoData <- GRanges(all_gene)
library(ChIPpeakAnno)
library(TxDb.Athaliana.BioMart.plantsmart28)
library(org.At.tair.db)
library(ensembldb)
txdb <- TxDb.Athaliana.BioMart.plantsmart28
anno <- annotatePeakInBatch(resUnlisted, AnnotationData=annoData, 
                  output="overlapping", 
                  FeatureLocForDistance="TSS",
                  bindingRegion=c(-2000, 300))
anno$symbol <- xget(anno$feature, org.At.tairSYMBOL)
edb <- EnsDb("EnsDb.Athaliana.vAtRTD2.sqlite")
peakAnno <- annotatePeak(resUnlisted, tssRegion=c(-3000, 3000),
                         TxDb=txdb, annoDb="org.At.tair.db")
peakAnno.edb <- annotatePeak(resUnlisted, tssRegion=c(-3000, 3000),
                             TxDb=txdb, annoDb="org.At.tair.db")
plotAnnoPie(peakAnno)
```

```{bash, eval=FALSE}

#installation 
#apart from meme

conda install -c conda-forge -c bioconda -c defaults prokka

#for this one nomodel works coz of the reason provided in the github link 
# http://seqanswers.com/forums/showthread.php?t=69769     ;) nice numbers in the link


#motif analysis required fasta files, could not really find packages to convert bed files to fasta so used this
#script found on stackexchange
#https://bioinformatics.stackexchange.com/questions/5368/how-to-convert-bed-file-to-fasta-file

perl bed2faidxsta.pl align_peaks_PE_summits.bed peaks_PE.fa
#perform motif analysis on peaks 

```

```{r}
#useless but adding, just coz
library(tidyverse)
library(microseq)
#BiocManager::install("BSgenome.Athaliana.TAIR.TAIR9")
library(BSgenome.Athaliana.TAIR.TAIR9)
genome <- BSgenome.Athaliana.TAIR.TAIR9
summitSeq = get_peak_summit_seq(file = "peaks_align_peaks.narrowPeak", peakFormat = "narrowPeak",genome = genome)
# CRR0 <- readFastq('CRR242640.fq.gz')
```

```{r}
# https://rdrr.io/github/lakhanp1/chipmine/man/get_peak_summit_seq.html

#install.packages("remotes")
#remotes::install_github("lakhanp1/chipmine")
```
# Authors and contributions

Authors: Alice Zhang (52721552) and Abhishek Dhir (87603866)

Contributions: Alice and Abhishek worked together synchronously and asynchronously on the project.